example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java

```diff
@@ -112,6 +115,36 @@ public void shouldReturnAllRecipes() {
         assertThat(recipeList, hasItems(recipeC));
     }
 
+    @Test
+    public void removeTest() {
+        List<Recipe> recipeList = recipeRepository.findAll();
+
+        assertThat(recipeList, hasSize(3));
+        assertThat(recipeList, hasItems(recipeA));
+        assertThat(recipeList, hasItems(recipeB));
+        assertThat(recipeList, hasItems(recipeC));
+
+        Recipe recipeAA = recipeRepository.findBy(recipeA.getId());
+        recipeRepository.remove(recipeAA);
+        Recipe recipeAAA = recipeRepository.findBy(recipeA.getId());
+        Assert.assertNull(recipeAAA);
+
+        Recipe recipeBB = recipeRepository.findBy(recipeB.getId());
+        recipeRepository.remove(recipeBB);
+        Recipe recipeBBB = recipeRepository.findBy(recipeB.getId());
+        Assert.assertNull(recipeBBB);
+
+        Recipe recipeCC = recipeRepository.findBy(recipeC.getId());
+        recipeRepository.remove(recipeCC);
+        Recipe recipeCCC = recipeRepository.findBy(recipeC.getId());
+        Assert.assertNull(recipeCCC);
+
+
+        List<Recipe> recipeList2 = recipeRepository.findAll();
```

Comment:
suggest renaming the variable to remainingRecipes or something like that. Maybe even use initialRecipes as the first variable.